New 37 comment

SemyonHoyrish opened the pull request.

SemyonHoyrish committed:
fix Makefile

SemyonHoyrish committed:
Implemented Comment gRPC service

SemyonHoyrish committed:
internal-api-protos pull request merged

Rogon1256 commented on endpoints/comment.go:
```diff
@@ -0,0 +1,230 @@
+	"google.golang.org/grpc"
+)
+
+const COMMENTS_GET_MAX_LIMIT uint32 = 100
```
Можно ли уменьшить int?

Rogon1256 commented on endpoints/comment.go:
```diff
@@ -0,0 +1,230 @@
+
+	for _, model := range commentModels {
+		comments = append(comments, &pb.CommentResponse{
+			ID:        model.ID,
+			Text:      model.Text,
+			VideoID:   model.VideoID,
+			CommentID: model.CommentID,
+			UserID:    model.UserID,
+			Likes:     model.Likes,
+			Dislikes:  model.Dislikes,
+			CreatedAt: uint64(model.CreatedAt.Unix()),
+			EditedAt:  uint64(model.EditedAt.Unix()),
+		})
```
Данная структура встречается второй раз  может вынести её в отдельную функцию?

Rogon1256 commented on endpoints/comment.go:
```diff
@@ -0,0 +1,230 @@
+
+	if result.RowsAffected != 1 {
+		return nil, errors.Errorf("Given user hadn't reacted to this video")
+	}
```
Если пользователь впервые реагирует на комментарий, его не будет в бд

Rogon1256 commented on endpoints/comment.go:
```diff
@@ -0,0 +1,230 @@
+
+	if result.RowsAffected != 1 {
+		return nil, errors.Errorf("Given user hadn't reacted to this video")
+	}
```
Тут надо возвращать не ошибку, а лайк=фолз, дизлайк=фолз

SemyonHoyrish committed:
Some changes in Comment gRPC service:

If user hadn't reacted to the given comment, we just return {false, false}
If user want to react to the given comment, we support it in update request
Also add transaction and raw sql query to update `likes` and `dislikes` fields in comment entity

SemyonHoyrish committed:
fix
